Fix some tests in test_c10d_functional_native.py

yifuwang opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #119229
* #119104
* #119103
* #118986
* __->__ #119102

Summary:
This PR fixes a few tests that were broken because `empty` became `empty_strided_cuda` in the generate code.

Also changed some _c10d_functional calls to funcol calls so add coverage to tracing.

cc @mrshenli @pritamdamania87 @zhaojuanmao @satgera @rohan-varma @gqchen @aazzolini @osalpekar @jiayisuse @H-Huang @kwen2501 @awgu @penguinwu @fegin @XilunWu @wanchaol @fduwjj @wz337 @tianyu-l @wconstab @yf225 @voznesenskym @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @chenyang78 @kadeng @muchulee8 @aakhundov @ColinPeppler

Commit:
Fix some tests in test_c10d_functional_native.py

Summary:
This PR fixes a few tests that were broken because `empty` became `empty_strided_cuda` in the generate code.

Also changed some _c10d_functional calls to funcol calls so add coverage to tracing.

[ghstack-poisoned]

Commit:
Update on "Fix some tests in test_c10d_functional_native.py"


Summary:
This PR fixes a few tests that were broken because `empty` became `empty_strided_cuda` in the generate code.

Also changed some _c10d_functional calls to funcol calls so add coverage to tracing.

[ghstack-poisoned]

Commit:
Update on "Fix some tests in test_c10d_functional_native.py"


Summary:
This PR fixes a few tests that were broken because `empty` became `empty_strided_cuda` in the generate code.

Also changed some _c10d_functional calls to funcol calls so add coverage to tracing.

[ghstack-poisoned]

Commit:
Update on "Fix some tests in test_c10d_functional_native.py"


Summary:
This PR fixes a few tests that were broken because `empty` became `empty_strided_cuda` in the generate code.

Also changed some _c10d_functional calls to funcol calls so add coverage to tracing.

[ghstack-poisoned]

Commit:
Update on "Fix some tests in test_c10d_functional_native.py"


Summary:
This PR fixes a few tests that were broken because `empty` became `empty_strided_cuda` in the generate code.

Also changed some _c10d_functional calls to funcol calls so add coverage to tracing.

[ghstack-poisoned]

wanchaol commented on test/distributed/test_c10d_functional_native.py:
```diff
@@ -335,11 +335,11 @@ def test_inductor_all_reduce_single(self):
             # Expect in-place with inductor allocated buf
-            ar0 = torch.ops._c10d_functional.all_reduce(buf0, "avg", "default")
-            ar0 = torch.ops._c10d_functional.wait_tensor(ar0)
+            ar0 = funcol.all_reduce(buf0, "avg", "default")
```
I assume we triggered the env var to use native funcol somewhere in this test?

wanchaol reviewed (approved):
lgtm

Commit:
Update on "Fix some tests in test_c10d_functional_native.py"


Summary:
This PR fixes a few tests that were broken because `empty` became `empty_strided_cuda` in the generate code.

Also changed some _c10d_functional calls to funcol calls so add coverage to tracing.

[ghstack-poisoned]

Commit:
Update on "Fix some tests in test_c10d_functional_native.py"


Summary:
This PR fixes a few tests that were broken because `empty` became `empty_strided_cuda` in the generate code.

Also changed some _c10d_functional calls to funcol calls so add coverage to tracing.

[ghstack-poisoned]

Commit:
Update on "Fix some tests in test_c10d_functional_native.py"


Summary:
This PR fixes a few tests that were broken because `empty` became `empty_strided_cuda` in the generate code.

Also changed some _c10d_functional calls to funcol calls so add coverage to tracing.

[ghstack-poisoned]

Commit:
Update on "Fix some tests in test_c10d_functional_native.py"


Summary:
This PR fixes a few tests that were broken because `empty` became `empty_strided_cuda` in the generate code.

Also changed some _c10d_functional calls to funcol calls so add coverage to tracing.

cc mrshenli pritamdamania87 zhaojuanmao satgera rohan-varma gqchen aazzolini osalpekar jiayisuse H-Huang kwen2501 awgu penguinwu fegin XilunWu wanchaol fduwjj wz337 tianyu-l wconstab yf225 voznesenskym EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx peterbell10 ipiszy chenyang78 kadeng muchulee8 aakhundov ColinPeppler

[ghstack-poisoned]

Commit:
Update on "Fix some tests in test_c10d_functional_native.py"


Summary:
This PR fixes a few tests that were broken because `empty` became `empty_strided_cuda` in the generate code.

Also changed some _c10d_functional calls to funcol calls so add coverage to tracing.

cc mrshenli pritamdamania87 zhaojuanmao satgera rohan-varma gqchen aazzolini osalpekar jiayisuse H-Huang kwen2501 awgu penguinwu fegin XilunWu wanchaol fduwjj wz337 tianyu-l wconstab yf225 voznesenskym EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx peterbell10 ipiszy chenyang78 kadeng muchulee8 aakhundov ColinPeppler

[ghstack-poisoned]

yifuwang commented:
@pytorchbot merge
